pymc3/variational/advi.py

```diff
@@ -135,11 +140,23 @@ def advi(vars=None, start=None, model=None, n=5000, accurate_elbo=False,
     updates = optimizer(loss=-1 * elbo, param=[uw_shared])
     f = theano.function([], [uw_shared, elbo], updates=updates, mode=mode)
 
+    # For tracking convergence of ELBO
+    window_size = int(max(0.1 * n // eval_elbo, 2.0))
+    circ_buff = deque([], maxlen=window_size)
+    converged = False
+
     # Optimization loop
     elbos = np.empty(n)
     try:
         progress = trange(n)
-        for i in progress:
+        uw_i, elbo_current = f()
+        if np.isnan(elbo_current):
+            raise FloatingPointError('NaN occurred in ADVI optimization.')
+        elbos[0] = elbo_current
+        progress.update(1)
+        i = 1
+        while not converged and i < n:
```

Comment:
This is definitely easier. I just never know what someone's aversion to breaks might be. I will change this.